server/src/main/java/org/elasticsearch/action/admin/indices/shrink/TransportResizeAction.java

```diff
@@ -122,31 +132,25 @@ protected void masterOperation(final ResizeRequest resizeRequest,
     }
 
     // static for unittesting this method
-    static CreateIndexClusterStateUpdateRequest prepareCreateIndexRequest(final ResizeRequest resizeRequest,
+    static CreateIndexClusterStateUpdateRequest prepareCreateIndexRequest(final ResizeRequest request,
                                                                           final ClusterState state,
                                                                           final IntFunction<DocsStats> perShardDocStats,
                                                                           String sourceIndexName,
                                                                           String targetIndexName) {
-        final CreateIndexRequest targetIndex = resizeRequest.getTargetIndexRequest();
         final IndexMetadata metadata = state.metadata().index(sourceIndexName);
         if (metadata == null) {
             throw new IndexNotFoundException(sourceIndexName);
         }
-        final Settings.Builder targetIndexSettingsBuilder = Settings.builder().put(targetIndex.settings())
-            .normalizePrefix(IndexMetadata.INDEX_SETTING_PREFIX);
-        targetIndexSettingsBuilder.remove(IndexMetadata.SETTING_HISTORY_UUID);
-        final Settings targetIndexSettings = targetIndexSettingsBuilder.build();
-        final int numShards;
-        if (IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.exists(targetIndexSettings)) {
-            numShards = IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.get(targetIndexSettings);
-        } else {
-            assert resizeRequest.getResizeType() == ResizeType.SHRINK : "split must specify the number of shards explicitly";
-            numShards = 1;
-        }
+        int currentNumShards = metadata.getNumberOfShards();
 
-        for (int i = 0; i < numShards; i++) {
-            if (resizeRequest.getResizeType() == ResizeType.SHRINK) {
-                Set<ShardId> shardIds = IndexMetadata.selectShrinkShards(i, metadata, numShards);
+        final Settings targetSettings = Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, request.newNumShards())
+            .build();
+        final int targetNumShards = request.newNumShards();
+        boolean shrink = currentNumShards > targetNumShards;
```

Comment:
I wonder if we should turn equal case into no-op - looks like it will go a split way? 

Maybe not be an issue but redundant